src/shared/hooks/useLatest.ts

```diff
@@ -0,0 +1,11 @@
+import { useLayoutEffect, useRef } from 'react';
+
+export function useLatest<T>(value: T) {
```

Comment:
You wrote a function, using `function` style, but in useThemeContext.ts used `const`. Choose and keep style of function declaring